kitsune/wiki/views.py

```diff
@@ -1453,3 +1479,116 @@ def what_links_here(request, document_slug):
     }
 
     return render(request, 'wiki/what_links_here.html', c)
+
+
+def get_fallback_locale(doc, request):
+    """Get best fallback local based on locale Mapping"""
+
+    # Get localized article versions with current revisions
+    translated_locales = (doc.translations.all()
+                                          .values_list('locale', flat=True)
+                                          .exclude(current_revision=None))
+    # Get the ACCEPT_LANGUAGE header from the user's request
+    browser_language_code = request.META.get('HTTP_ACCEPT_LANGUAGE', None)
+    if browser_language_code is not None:
+        header_locales = parse_accept_lang_header(browser_language_code)
+    else:
+        header_locales = []
+    # Get the locale mapping from the Kitsune settings
+    NON_SUPPORTED_LOCALES = settings.NON_SUPPORTED_LOCALES
+
+    # Check if the document is localized into any of the locales listed in
+    # the ACCEPT_LANGUAGE header
+    for locale in header_locales:
+        if locale in translated_locales:
+            fallback_locale = locale
+            break
+    else:
+        # If there are no versions matching the locales listed in the ACCEPT_LANGUAGE header,
+        # check for fallback locales defined for locales listed in the ACCEPT_LANGUAGE header.
+        # If the document is available in a fallback locale, show that locale
+        # The fallback locales are defined in the FALLBACK_LOCALES array in kitsune/wiki/config.py
+        for locale in header_locales:
+            if NON_SUPPORTED_LOCALES.get(locale) in translated_locales:
+                fallback_locale = NON_SUPPORTED_LOCALES[locale]
+                break
```

Comment:
You should apply this pattern of returning early to the rest of the function as well. It should greatly improve the clarity.